plugin/pkg/timeouts/timeouts.go

```diff
@@ -0,0 +1,30 @@
+package timeouts
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+// NewTimeoutFromArg returns a time.Duration from a configuration argument
+// (string) which has come from the Corefile. The argument has some basic
+// validation applied before returning a time.Duration.
+func NewTimeoutFromArg(arg string) (time.Duration, error) {
```

Comment:
why do we want this externalized? can we rather have it only as part of `plugin/timeouts` package?

Reply:
if we want this externalized I think we should try to think of better name of a package, this is quite common duration parsing, not necessarily timeout
```suggestion
func NewDurationFromArg(arg string) (time.Duration, error) {
```

Reply:
I have no good reason to externalise it. Looking at the TLS plugin it looked like anything beyond very basic parsing was externalised. I would be happy to try and bring it together if it was felt it would be better.

Reply:
I think I would leave it (mostly) externalised but rename to durations. It would do the basic parsing and default durations passed as integers to seconds as it does currently.

The check for "max" and "min" durations will be moved into the `plugin/timeouts` bit since durations parsed in other parts of the code may not have such constraints.

Would you be happy with that approach? i will convert the tests to table tests as well

Reply:
Sounds good 👍